pkg/runner/verify_test.go

```
@@ -49,3 +50,34 @@ func TestConnectVerificationBoundedAndVerifies(t *testing.T) {
 	require.True(t, scanner.ScanResults.IPHasPort("127.0.0.1", openP), "open port must survive verification")
 	require.False(t, scanner.ScanResults.IPHasPort("127.0.0.1", closedP), "closed port must be dropped by verification")
 }
+
+func TestConnectVerificationKeepsCappedHosts(t *testing.T) {
+	ln, err := net.Listen("tcp", "127.0.0.1:0")
```

golangci-lint (2.12.2)

[error] net.Listen must not be called. use (*net.ListenConfig).Listen

(noctx)